config/yaml.go

```diff
@@ -329,36 +325,34 @@ func unmarshalYAMLValue(reader io.ReadCloser, value interface{}) error {
 //
 // TODO: what if someone wanted a literal ${FOO} in config? need a small escape hatch
 func interpolateEnvVars(data []byte) ([]byte, error) {
-	var errs []string
-	data = _interpolationRegex.ReplaceAllFunc(data, func(in []byte) []byte {
-		// remove ${ and }, just look inside the braces
-		valAndMaybeDefault := in[2 : len(in)-1]
-
-		// ${VALUE:DEFAULT} -> VALUE, DEFAULT
-		// ${VALUE} -> VALUE, empty default
-		split := bytes.Split(valAndMaybeDefault, _envSeparator)
-		key := string(split[0])
-
-		if envVal, ok := os.LookupEnv(key); ok {
-			return []byte(envVal)
+	// Is this conversion ok?
```

Comment:
Was just a note from myself to myself to make sure it's not too wasteful. There are a lot of similar APIs offered between `[]byte` and `string`, so there may be a way to do it without this conversion at all, but the again, that might not be all that necessary from the get go, especially since this only happens once